oak-jackrabbit-api/src/main/java/org/apache/jackrabbit/api/JackrabbitSession.java

```diff
@@ -268,6 +268,9 @@ public interface JackrabbitSession extends Session {
      */
     @Nullable 
     default Node getParentOrNull(@NotNull Item item) throws RepositoryException {
+        if (item instanceof JackrabbitNode) {
+            return ((JackrabbitNode) item).getParentOrNull();
+        }
```

Comment:
I'd put an "else" here (I don't like unbalanced "return"s, but that's probably just a matter of taste.